airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftS3StagingInsertDestinationAcceptanceTest.java

```diff
@@ -0,0 +1,150 @@
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.db.Database;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.CatalogHelpers;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.DestinationSyncMode;
+import io.airbyte.protocol.models.JsonSchemaType;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.List;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RedshiftS3StagingInsertDestinationAcceptanceTest extends RedshiftCopyDestinationAcceptanceTest {
+
+  public static final String DATASET_ID = Strings.addRandomSuffix("airbyte_tests", "_", 8);
+  private static final String TYPE = "type";
+  private ConfiguredAirbyteCatalog catalog;
+
+  private static final Instant NOW = Instant.now();
+  private static final String BOOKS_STREAM_NAME = "books_" + RandomStringUtils.randomAlphabetic(5);
+
+  private static final AirbyteMessage MESSAGE_BOOKS1 = new AirbyteMessage().withType(AirbyteMessage.Type.RECORD)
+          .withRecord(new AirbyteRecordMessage().withStream(BOOKS_STREAM_NAME)
+                  .withData(Jsons.jsonNode(ImmutableMap.builder().put("name", "Harry Potter").put("id", "10").build()))
+                  .withEmittedAt(NOW.toEpochMilli()));
+  private static final AirbyteMessage MESSAGE_BOOKS2 = new AirbyteMessage().withType(AirbyteMessage.Type.RECORD)
+          .withRecord(new AirbyteRecordMessage().withStream(BOOKS_STREAM_NAME)
+                  .withData(Jsons.jsonNode(ImmutableMap.builder().put("name", "The Great Gatsby").put("id", "30").build()))
+                  .withEmittedAt(NOW.toEpochMilli()));
+  private static final AirbyteMessage BOOKS_IN_THE_DB = new AirbyteMessage().withType(AirbyteMessage.Type.RECORD)
+          .withRecord(new AirbyteRecordMessage().withStream(BOOKS_STREAM_NAME)
+                  .withData(Jsons.jsonNode(ImmutableMap.builder().put("name", "Brave New World").put("id", "1").build()))
+                  .withEmittedAt(NOW.toEpochMilli()));
+
+  private static final AirbyteMessage MESSAGE_STATE = new AirbyteMessage().withType(AirbyteMessage.Type.STATE)
+          .withState(new AirbyteStateMessage().withData(Jsons.jsonNode(ImmutableMap.builder().put("checkpoint", "now!").build())));
+
+  public JsonNode getStaticConfig() {
+    return Jsons.deserialize(IOs.readFile(Path.of("secrets/config.json")));
+  }
+
+  @BeforeEach
+  void setup(){
+    MESSAGE_BOOKS1.getRecord().setNamespace(DATASET_ID);
```

Comment:
nitpick: why does this need to happen in the beforeeach?

Reply:
For now, only one test in each class exists could be removed. Thanks. Updated.